pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -540,3 +550,57 @@ func hostDevicesOvercommitNeeded(oldVM, newVM *kubevirtv1.VirtualMachine) bool {
 func isDedicatedCPU(vm *kubevirtv1.VirtualMachine) bool {
 	return vm.Spec.Template.Spec.Domain.CPU != nil && vm.Spec.Template.Spec.Domain.CPU.DedicatedCPUPlacement
 }
+
+func (m *vmMutator) ensureMaintenanceModeStrategyIsSet(vm *kubevirtv1.VirtualMachine, patchOps types.PatchOps) (types.PatchOps, error) {
+	// If the maintenante-mode strategy label is not set, add it with the default
+	// value of `Migrate`.
+	labels := vm.ObjectMeta.Labels
+	strategy, ok := labels[util.LabelMaintainModeStrategy]
+	if !ok {
+		strategy = util.MaintainModeStrategyMigrate
+		logrus.WithFields(logrus.Fields{
+			"namespace": vm.Namespace,
+			"name":      vm.Name,
+		}).Infof("no maintenance mode strategy specified for VM, applying default strategy `Migrate`")
+		patch, err := json.Marshal(
+			util.PatchStringValue{
+				Op:    "replace",
+				Path:  "/metadata/labels/harvesterhci.io~1maintain-mode-strategy", // `~1` is escaped encoding for `/` in JSON
+				Value: strategy,
+			},
+		)
+		if err != nil {
+			return patchOps, err
+		}
+
+		patchOps = append(patchOps, string(patch[:]))
+	}
+
+	// If the value under .spec.template.metadata.labels does not match the
```

Comment:
we need to differentiate the `create` and `update` case:

for `create`, a wrong strategy should be rejected to tell user the clear error, then he will notice and adjust it;

for `update`, it may come from legacy vm, adjust the wrong value is user friendly

Reply:
The mutator will not reject anything, since it's not its job.

The mutator will just make sure that the label is set (or it will set it itself to the default) and that the label in the `.spec.template` matches the label in `.metadata`.

If the label is set, but to an invalid value, the mutator will not change it, because this would be unexpected by the user.
However, the validator will later reject the operation, if the label is invalid.